File: src/main/java/com/jesperancinha/atm/finder/ATMService.java
package com.jesperancinha.atm.finder;

import com.jesperancinha.atm.finder.service.ATMLocatorService;
import lombok.NoArgsConstructor;
import org.apache.camel.BeanInject;

import java.io.IOException;

@NoArgsConstructor
public class ATMService {

    @BeanInject
    private ATMLocatorService atmLocatorService;

    public ATMProvider getATMProvider(String city) throws IOException {
        return ATMProvider.builder().atmMachines(atmLocatorService.getAtmPerCity(city.toUpperCase()))
                .city(city).build();
    }
}

File: jeorg-camel-app-atm/src/main/kotlin/org/jesperancinha/atm/finder/camel/AtmService.kt
package org.jesperancinha.atm.finder.camel

import org.jesperancinha.atm.finder.service.AtmLocatorService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.io.IOException
import java.util.*

@Service
class AtmService(
    @Autowired
    val atmLocatorService: AtmLocatorService
) {

    @Throws(IOException::class)
    fun getATMProvider(city: String): AtmProvider {
        return AtmProvider(
            atmMachines = atmLocatorService.getAtmPerCity(city.uppercase(Locale.getDefault())),
            city = city
        )
    }
}